File: RemoteInstaller/app/src/main/java/remoteapkinstallerclient/com/br/remoteinstaller/view/Activities/SettingsActivity.java
package remoteapkinstallerclient.com.br.remoteinstaller.view.Activities;

import android.os.Bundle;
import android.preference.EditTextPreference;
import android.preference.ListPreference;
import android.preference.Preference;
import android.preference.PreferenceFragment;
import android.support.v7.app.AppCompatActivity;
import android.view.MenuItem;

import remoteapkinstallerclient.com.br.remoteinstaller.R;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getFragmentManager().beginTransaction().replace(android.R.id.content, new MyPreferenceFragment()).commit();

        if (getSupportActionBar() != null) {
            getSupportActionBar().setHomeButtonEnabled(true);
            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        }

    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

    public static class MyPreferenceFragment extends PreferenceFragment implements Preference.OnPreferenceChangeListener {
        @Override
        public void onCreate(Bundle savedInstanceState) {
            super.onCreate(savedInstanceState);
            addPreferencesFromResource(R.xml.preferences);
            updateSummaryForPreference("preferences_ip");
            updateSummaryForPreference("preferences_port");
        }

        @Override
        public boolean onPreferenceChange(Preference preference, Object o) {
            String value = o.toString();
            if (preference instanceof EditTextPreference) {
                EditTextPreference etp = (EditTextPreference) preference;
                preference.setSummary(etp.getText());
            }
            return true;
        }

        private void updateSummaryForPreference(String preferenceKey) {
            findPreference(preferenceKey).setOnPreferenceChangeListener(this);
        }

    }

}

File: RemoteInstaller/app/src/main/java/remoteapkinstallerclient/com/br/remoteinstaller/view/Activities/SettingsActivity.kt
package remoteapkinstallerclient.com.br.remoteinstaller.view.Activities

import android.os.Bundle
import android.preference.EditTextPreference
import android.preference.ListPreference
import android.preference.Preference
import android.preference.PreferenceFragment
import android.support.v7.app.AppCompatActivity
import android.view.MenuItem

import remoteapkinstallerclient.com.br.remoteinstaller.R

class SettingsActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        fragmentManager.beginTransaction().replace(android.R.id.content, MyPreferenceFragment()).commit()

        if (supportActionBar != null) {
            supportActionBar!!.setHomeButtonEnabled(true)
            supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
        }
        return super.onOptionsItemSelected(item)
    }

    class MyPreferenceFragment : PreferenceFragment(), Preference.OnPreferenceChangeListener {
        override fun onCreate(savedInstanceState: Bundle?) {
            super.onCreate(savedInstanceState)
            addPreferencesFromResource(R.xml.preferences)
            updateSummaryForPreference("preferences_ip")
            updateSummaryForPreference("preferences_port")
        }

        override fun onPreferenceChange(preference: Preference, o: Any): Boolean {
            val value = o.toString()
            (preference as? EditTextPreference)?.setSummary(preference.text)
            return true
        }

        private fun updateSummaryForPreference(preferenceKey: String) {
            findPreference(preferenceKey).onPreferenceChangeListener = this
        }

    }

}
